libcontainer/init_linux.go

```diff
@@ -390,7 +390,16 @@ func setupRlimits(limits []configs.Rlimit, pid int) error {
 func setOomScoreAdj(oomScoreAdj int, pid int) error {
 	path := fmt.Sprintf("/proc/%d/oom_score_adj", pid)
 
-	return ioutil.WriteFile(path, []byte(strconv.Itoa(oomScoreAdj)), 0600)
+	// If there's a permission error, continue. In unprivileged LXD containers, this isn't allowed.
+	// See https://github.com/lxc/lxd/issues/2994 .
+	if err := ioutil.WriteFile(path, []byte(strconv.Itoa(oomScoreAdj)), 0600); err != nil {
+		if os.IsPermission(err) {
```

Comment:
I think we want to check if we are running in a userns or not for error handling.

If we are in a userns, then don't even warn.
If we are not, then return the error regardless of an eperm